### DISCUSSION TITLE
Phone Numbers are not properly displayed

### DISCUSSION THREAD

**Discussion by cateas** [deepgram/community#261] [General help] [answered]
### Which Deepgram product are you using?

Deepgram SDKs

### Details

We have noticed in multiple call recordings that the phone number is not properly returned in the transcription.

For example in the call recording we have: 702481 98 54, but Deepgram returns 07:02 4 08/01/9854.

Another example: 5406876106, but Deepgram returns 5 04:06 876 01:06. Could you please let us know if there is a parameter which could help us to get a proper format for the phone call numbers extract from transcription? 

I can share some example of files via email. Please let me know if this is necessary.

Thank you!

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

POST /v1/listen?multichannel=true&smart_format=true&utterances=true&tier=nova&model=general&utt_split=1.5&detect_language=true

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

08603702-4672-44ea-9bca-e60719d61a15

**Comment by jjmaldonis** [comment, marked as answer]
Hey @cateas thanks for the report with the detailed examples. We released a new version of smart formatting that is _much_ better than the old version, and you found an edge case where the old smart formatter was being used. This is a bug on our end and we are now in the process of getting it fixed. So there's nothing to do on your end, we'll get this fixed and the results will improve dramatically.

**Reply by cateas** [reply]
Great. Thank you!
Could you please notify us when it is fixed. Meanwhile we are using numbers, since for us the phone numbers are quite important.

Thank you!

**Reply by jjmaldonis** [reply]
Yes we will let you know when it's fixed!

**Reply by shirgoldbird** [reply]
@cateas This should be fixed! When you're using our Nova general model, and our language detection detects your file is in English, you'll receive the improved smart formatting output. Please test out your original request and let us know how it looks.

**Reply by cateas** [reply]
Great! Thank you very much. I did couple of tests and it seems it works good.
I will ask our developer to apply the change and we will monitor it.

**Reply by rrichc** [reply]
Hi @jjmaldonis , was this new version of smart formatting fixed in all models or just the general model? I'm finding that the phonecall model is mishandling phone numbers very similarly to the original post quite frequently

**Reply by shirgoldbird** [reply]
@rrichc, we have an improved version of Smart Formatting available for Enhanced general model and our Nova general model. Other models are currently still using the older version of Smart Formatting.

**Reply by rrichc** [reply]
@shirgoldbird Understood, thank you

**Comment by cateas** [comment]
@jjmaldonis I have noticed some inconsistencies and missing words for the case when we use smart-formatting and when we use numerals.

For example:

**Smart Formatting:**
Thank you for calling the law office of Test. How may I assist you? Okay. Perfect. No worries at And since they are unavailable, I will have someone from the office give you a callback as soon as possible. May I please confirm it was Abc. Right? Thank you, Abc. May I please have your last name as well? Oh, I have that here for you. Okay. Perfect. Is that gonna be 1 11 and then 1111111. Perfect. And then abcdef, so I don't butcher that as well. And then the client that this is in reference to. Okay. I'm gonna confirm that on my end, a b c d. And then day, is that just c d? And then what was the rest? a for Adele. Okay. Perfect. And was this for an update regarding this, or what was the call in regards to for this? Just a status update in general. Okay. I'll go ahead and have them call you back on their direct as soon as possible. Okay? Confirming claim number as 0 0000000000 dash 1. Perfect. You're all set, Abc. I'll go ahead and get this over and have them call you back as soon as possible for the status update. Okay? Thank you. Have a great day.

**Numerals:**
Thank you for calling the law office of Test. How may I assist you? Okay. Perfect. No worries at all. And since they are unavailable, I'll have someone from the office give you a callback as soon as possible. May I please confirm it was Abc. Right? Thank you, Abc. May I please have your last name as well? Oh, I have that here for you. Okay. Perfect. Is that gonna be 1 1 1 and then 1 1 1 1 1 1 1. Perfect. And then a b c d e f, so I don't butcher that as well. And then the client that this is in reference to. Okay. I'm gonna confirm that on my end, a b c d, and then day. Is that just c d? And then what was the rest? **Alright. E e l a, and then Veka, v e g a, Aguilar, a g u I l e r** a for Adel. Okay. Perfect. And was this for an update regarding this or what was the call in regards to for this? Just a status update in general. Okay. I'll go ahead and have them call you back on their direct as soon as possible. Okay? Yeah. I'm ready. Confirming claim number as 0 0 0 0 0 0 0 0 0 0 0 dash 1? Perfect. You're all set, Abc. I'll go ahead and get this over and have them call you back as soon as possible for the status update. Okay? Thank you. Have a great day.

I will share the audio file via email. Could you please check this ?

**Reply by jjmaldonis** [reply]
Thanks @cateas, I just checked to see if the same issue occurs for Nova 2, our latest model, and the smart formatted transcript correctly contained the sentence. Have you tried Nova 2? / What model are you using? Alternatively, do you have a request ID?

**Reply by cateas** [reply]
@jjmaldonis we do not have access to nova-2 yet. I have sent a request for this. 

We are using nova (original model).
We plan to move to nova-2, but we also need Spanish :).

**Reply by jjmaldonis** [reply]
Got it, that makes sense. It's unlikely the formatting issue will be fixed for Nova 1 since it is already fixed in Nova 2. Nova 1 will be effectively out of date as soon as Spanish is available. Once you receive access to Nova 2, will you be able to send your English files to Nova 2 and your Spanish files to Nova 1? I know you were using detect-language a couple months ago - has that changed?

P.S. The work to restrict the languages that can be detected by `detect_language=true` is underway. :)

**Reply by cateas** [reply]
Well, we are still using detect language. Unfortunately, we do not know if the call is in English or Spanish.
